contracts/ColonyAuthority.sol

```diff
@@ -17,64 +17,81 @@
-    setFounderRoleCapability(colony, "addDomain(uint256)");
-    // Add task
-    setAdminRoleCapability(colony, "makeTask(bytes32,uint256,uint256,uint256)");
-    setFounderRoleCapability(colony, "makeTask(bytes32,uint256,uint256,uint256)");
-    // Start next reward payout
-    setAdminRoleCapability(colony, "startNextRewardPayout(address,bytes,bytes,uint256,bytes32[])");
-    setFounderRoleCapability(colony, "startNextRewardPayout(address,bytes,bytes,uint256,bytes32[])");
-    // Set admin
-    setAdminRoleCapability(colony, "setAdminRole(address)");
-    setFounderRoleCapability(colony, "setAdminRole(address)");
+
+    // Add permissions for the Administration role
+    setAdministrationCapability(colony, "makeTask(uint256,uint256,bytes32,uint256,uint256,uint256)");
+    // Only for admins in root domain
+    setAdministrationCapability(colony, "startNextRewardPayout(address,bytes,bytes,uint256,bytes32[])");
+
+    // Add permissions for the Funding role
+    setFundingCapability(colony, "moveFundsBetweenPots(uint256,uint256,uint256,uint256,uint256,uint256,address)");
+
+    // Add permissions for the Architecture role
+    setArchitectureCapability(colony, "addDomain(uint256,uint256,uint256)");
+    setArchitectureSubdomainCapability(colony, "setFundingRole(uint256,uint256,address,uint256,bool)");
+    setArchitectureSubdomainCapability(colony, "setAdministrationRole(uint256,uint256,address,uint256,bool)");
+    setArchitectureSubdomainCapability(colony, "setArchitectureRole(uint256,uint256,address,uint256,bool)");
+
+    // Add permissions for the Root role
+    setRootCapability(colony, "setFundingRole(uint256,uint256,address,uint256,bool)");
+    setRootCapability(colony, "setAdministrationRole(uint256,uint256,address,uint256,bool)");
+    setRootCapability(colony, "setArchitectureRole(uint256,uint256,address,uint256,bool)");
+    setRootCapability(colony, "setRootRole(address,bool)");
+
+    setRootCapability(colony, "setRecoveryRole(address)");
+    setRootCapability(colony, "removeRecoveryRole(address)");
+
+    setRootCapability(colony, "bootstrapColony(address[],int256[])");
+    setRootCapability(colony, "registerColonyLabel(string,string)");
+    setRootCapability(colony, "addNetworkColonyVersion(uint256,address)");
+    setRootCapability(colony, "setNetworkFeeInverse(uint256)");
+    setRootCapability(colony, "setRewardInverse(uint256)");
+    setRootCapability(colony, "upgrade(uint256)");
+    setRootCapability(colony, "mintTokens(uint256)");
+    setRootCapability(colony, "addGlobalSkill(uint256)");
+    setRootCapability(colony, "startNextRewardPayout(address,bytes,bytes,uint256,bytes32[])");
   }
 
-  function setFounderRoleCapability(address colony, bytes memory sig) private {
-    bytes4 functionSig = bytes4(keccak256(sig));
-    setRoleCapability(founderRole, colony, functionSig, true);
+  function setAdministrationCapability(address colony, bytes memory sig) private {
```

Comment:
Seeing that the `colony` address is passed in all these functions yet we have it in the constructor, and considering the `ColonyAuthority` contract only ever works with a single colony, we can set the colony address as a property of the authority and reuse it in permissions management functions. It'll simplify the interface somewhat.

Reply:
That's fair, this contract definitely has some boilerplate we can remove.